src/extraction/FStar.Extraction.ML.Modul.fs

```diff
@@ -670,6 +670,34 @@ let maybe_register_plugin (g:env_t) (se:sigelt) : list<mlmodule1> =
 (*****************************************************************************)
 (* Extracting the top-level definitions in a module                          *)
 (*****************************************************************************)
+
+let arg_count e =
+  match e.expr with
+  | MLE_Fun (l, _) ->
+      List.length l
+  | _ ->
+      0
+
+let rec is_stackinline tcenv n t =
+  let t = N.normalize [ Env.Inlining; Env.EraseUniverses; Env.AllowUnboundUniverses; Env.UnfoldUntil S.delta_constant ] tcenv t in
+  let comp_is_stackinline = function
+    | { n = Comp { effect_name = e } } when string_of_lid e = "FStar.HyperStack.ST.StackInline" -> true
+    | _ -> false
+  in
+  match U.arrow_formals_comp t with
+  | args, comp ->
+      if List.length args > n then
+        // let f x: int -> int = g --- n = 1, length args = 2, as returned by arrow_formals_comp
+        // it must've been a Tot arrow
+        false
+      else if List.length args = n && comp_is_stackinline comp then
+        true
+      else
+        // JP: can this even happen?
```

Comment:
This might happen since in extracting to ML we erase the prefix of type arguments, if any, for a polymorphic function

Reply:
Can you try your example with a polymorphic identity stack inline function, or something like it?

Reply:
yeah indeed doesn't work... thanks for the counterexample